File: cpplib/src/presents/net/AuthResponseData.cpp
#include "presents/stable.h"
#include "AuthResponseData.h"

using namespace presents::net;

DEFINE_STREAMABLE("com.threerings.presents.net.AuthResponseData", AuthResponseData);

void AuthResponseData::readObject (ObjectInputStream& in)
{
    presents::dobj::DObject::readObject(in);
    code = in.readField< utf8 >();
}

void AuthResponseData::writeObject (ObjectOutputStream& out) const
{
    presents::dobj::DObject::writeObject(out);
    out.writeField(code);
}

File: core/src/main/java/com/threerings/presents/net/AuthResponseData.java
//
// Narya library - tools for developing networked games
// Copyright (C) 2002-2025 Three Rings Design, Inc., All Rights Reserved
// https://github.com/threerings/narya/blob/master/LICENSE

package com.threerings.presents.net;

import com.threerings.presents.dobj.DObject;

/**
 * An <code>AuthResponseData</code> object is communicated back to the
 * client along with an authentication response. It contains an indicator
 * of authentication success or failure along with bootstrap information
 * for the client.
 */
public class AuthResponseData extends DObject
{
    /** The constant used to indicate a successful authentication. */
    public static final String SUCCESS = "success";

    /**
     * Either the {@link #SUCCESS} constant or a reason code indicating
     * why the authentication failed.
     */
    public String code;
}
